src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java

```diff
@@ -0,0 +1,83 @@
+package ru.yandex.practicum.filmorate.controller;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.web.bind.annotation.*;
+import ru.yandex.practicum.filmorate.exception.ValidationException;
+import ru.yandex.practicum.filmorate.model.User;
+
+import java.time.LocalDate;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+@Slf4j
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    public static void validate(User user) throws ValidationException {
+        if (user.getEmail() == null || user.getEmail().isBlank()) {
+            throw new ValidationException("Электронная почта не может быть пустой");
+        }
+
+        if (!user.getEmail().contains("@")) {
+            throw new ValidationException("Электронная почта должна содержать символ @");
+        }
+
+        if (user.getLogin() == null || user.getLogin().isBlank()) {
+            throw new ValidationException("Логин не может быть пустым");
+        }
+
+        if (user.getLogin().contains(" ")) {
+            throw new ValidationException("Логин не может содержать пробелы");
+        }
+
+        if (user.getBirthday().isAfter(LocalDate.now())) {
+            throw new ValidationException("Дата рождения не может быть в будущем");
+        }
+    }
+
+    private final Map<Integer, User> users = new HashMap<>();
+
+    @PostMapping
+    public User addUser(@RequestBody User user) {
```

Comment:
можно добавить валидацию аннотациями
для этого достаточно добавить зависимость
```xml
<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-validation</artifactId>
</dependency>
```

Поставить Valid
```suggestion
    public User addUser(@RequestBody @Valid User user) {
```

и в классе User например над полем email поставить две аннотации
```java
@Email(message = "Указан некорректный формат почты")
@NotBlank(message = "Поле email не может быть пустым или null")
private String email;
```

Reply:
и если попробовать сделать post с некорректной почтой, например `@abc.com@`, то возникнет исключение, и в ответе придет сообщение об ошибке